contracts/AgentMech.sol

```diff
@@ -95,4 +139,23 @@ contract AgentMech is ERC721Mech {
     function getRequestsCount(address account) external view returns (uint256 requestsCount) {
         requestsCount = mapRequestsCounts[account];
     }
+
+    /// @dev Gets the set of undelivered request Ids.
+    /// @return requestIds Set of undelivered request Ids.
+    function getUndeliveredRequestIds() external view returns (uint256[] memory requestIds) {
```

Comment:
Do we need to worry about the number of undelivered requests being too large. Although unlikely, I think it may happen. Could that result in a big request to the rpc which would get dropped constantly?
I am thinking maybe it makes sense to either have this method take a maxNumReqs, or have another method that supports that, just to avoid this case.

Reply:
Good point. How about we slightly alter the method to allow to (optionally) set a batch size. Then you can get constant sized returns

Reply:
@DavidMinarsch that should do

Reply:
I've accounted for both size and offset, please take a look.